### PR TITLE
Enable and fix no-unused-expressions

### PR DESCRIPTION
**Pull request by jtbandes**
**User-Facing Changes**
None

### CODE REVIEW

**Comment by defunctzombie** [comment]
nice!

File: packages/studio-base/src/components/MessagePathSyntax/useCachedGetMessagePathDataItems.ts [hunk 1]
```diff
@@ -177,7 +177,6 @@ export function fillInGlobalVariablesInPath(
         return { ...messagePathPart, value };
       }
 
-      messagePathPart.type;
```
Comment (jtbandes) [inline packages/studio-base/src/components/MessagePathSyntax/useCachedGetMessagePathDataItems.ts:180]: This was not random – in Flow it used to be a type assertion to validate the type refinement after the above conditional. Probably was removed at some point automatically by flow-to-ts or eslint.

File: packages/studio-base/src/components/AccountSettingsSidebar/SigninForm.tsx [hunk 2]
```diff
@@ -84,7 +84,6 @@ export default function SigninForm(): JSX.Element {
     if (!deviceResponse) {
       return;
     }
-    deviceResponse?.id_token;
```
Comment (jtbandes) [inline packages/studio-base/src/components/AccountSettingsSidebar/SigninForm.tsx:87]: cc @defunctzombie - anything we should do here? actually I probably should've deleted the whole useEffect huh?

Comment (defunctzombie) [reply packages/studio-base/src/components/AccountSettingsSidebar/SigninForm.tsx:87]: yea - this must have been some code I left accidentally - doesn't seem used in this effect

Comment (jtbandes) [reply packages/studio-base/src/components/AccountSettingsSidebar/SigninForm.tsx:87]: But the effect also doesn’t do anything at all

**Pull request merged**